backend/main/server/controllers.go

```diff
@@ -512,16 +432,52 @@ func (a *App) updateProposal(w http.ResponseWriter, r *http.Request) {
 		}
 	}
 
-	p.Status = &payload.Status
-	p.Cid, err = helpers.pinJSONToIpfs(p)
+	if *payload.Proposal.Status == "draft" {
+		p = *payload.Proposal
+
+		if err := p.UpdateDraftProposal(a.DB); err != nil {
+			log.Error().Err(err).Msg("Error updating draft proposal")
+			respondWithError(w, errIncompleteRequest)
+			return
+		}
+
+		respondWithJSON(w, http.StatusOK, p)
+	} else {
+		p.Status = payload.Proposal.Status
```

Comment:
should add a check here that the status transition is valid. i.e.

1. "draft" -> "published"
2. "published" -> "cancelled"